ObjCTraining/AddPropertyInfoViewController.m

```diff
@@ -0,0 +1,331 @@
+//
+//  AddPropertyInfoViewController.m
+//  ObjCTraining
+//
+//  Created by RWuser on 25/03/16.
+//  Copyright © 2016 RWuser. All rights reserved.
+//
+#import <QuartzCore/QuartzCore.h>
+#import "AddPropertyInfoViewController.h"
+#import "ValidationManager.h"
+
+typedef NS_ENUM( NSInteger, InputFieldsEnum) {
+    Address1FieldTag = 1,
+    Address2FieldTag,
+    CityFieldTag,
+    StateFieldTag,
+    ZipFieldTag,
+    CountryFieldTag
+};
+
+@interface AddPropertyInfoViewController()
+@property UIView *activeField;
+@end
+
+@implementation AddPropertyInfoViewController
+BOOL isFieldEmpty = NO;
+NSString * emptyFieldErrorMessage = @"Please Enter Full Address";
+NSString * wrongFieldErrorMessage = @"Please Enter Correct Address";
+
+NSMutableDictionary *isFieldsEmpty;
+NSMutableDictionary *isFieldsWrong;
+
+- (void)viewWillAppear:(BOOL)animated {
+    self.address1TextField.delegate = self;
```

Comment:
Why do you set delegates in viewWillAppear? Will it be changed after pop/push this viewcontroller?